### ISSUE TITLE
How to successfully use Live Objects ?

### ISSUE THREAD

**Issue by eric-creekside** [redisson/redisson#5298]
I'm having difficulty with two different parts of the Live Objects feature of Redisson:

1. Storing a nested `Map<String, String>` as a field of an '@REntity`
2. Using a condition to `find()` instances that I know are stored.

Here's my live object class (I'm using Lombok, hence the @Getter, etc annotations):
```java
@RequiredArgsConstructor
@NoArgsConstructor(access = PROTECTED)
@REntity(codec = StringCodec.class)
public class Manifest {

	@RId(generator = UUIDGenerator.class)
	@Getter
	private String id;

	@RIndex
	@Getter
	private String jobId;

	@RIndex
	@Getter
	private String rootFolder;

	@RCascade(ALL)  // Not sure this is needed, it doesn't seem to make any difference
	private Map<String, String> entries = new HashMap<>();


	public void add(String path, String hash) {
		entries.put(path, hash);
	}

	@Transient
	public int getSize() {
		return entries.size();
	}

}
```

I'm using this code to initially create and store the live object:
```java
	RLiveObjectService liveObjects = redisson.getLiveObjectService();
	var manifest = new Manifest(jobID, rootPath);
	manifest = liveObjects.persist(manifest);
	liveObjects.asLiveObject(manifest).expire(Duration.ofMinutes(60));
```

Elsewhere I use this code to get the object by ID and add data to its `entries` map:
```java
	Manifest manifest = liveObjects.get(Manifest.class, id);
	manifest.add(path, hash);
```

Somewhere else, I use this code to find the object:
```java
	public static Condition jobIDEquals(String jobID) {
		return Conditions.eq("jobId", jobID);
	}

	Collection<Manifest> manifests = liveObjects.find(Manifest.class, jobIDEquals(jobID));
```

With the code above, the two problems are:
1. The `manifests` collection is empty, even though I can inspect my Redis server (using RedisInsight) and see the hashes stored there with matching `jobId` values.
2. If I load one of the objects directly by its `id`, the `entries` Map is always empty, _even after_ having added values to it. I see in RedisInsight that the value of `entries` is `org.redisson.RedissonReference@7623beea`, but there is no other key in Redis that corresponds to that reference.

What am I doing wrong?

**Comment by mrniko** [comment]
try to define getters and constructor explicitly

**Comment by eric-creekside** [comment]
That solved some of the problems, thanks. I strongly suggest you make this clear in the documentation and examples that Live Objects can not use getters/setters/constructors that are generated by byte-code tools. Lombok is very popular and commonly used. It could save other users a lot of time knowing to avoid using it in their @REntity objects.

**Comment by eric-creekside** [comment]
As I tested more, I was still seeing the situation where values I had added to my nested `entries` `Map<>` were not getting persisted, so when a later process loaded the objects with `find()`, `entries` was empty. I discovered that my `Manifest.add()` method had to use the getter to reference `entities`. Specifically, this code did *not* work:
```java
	public void add(String path, String hash) {
		entries.put(path, hash);
	}
```
but this code does work:
```java
	public void add(String path, String hash) {
		getEntries().put(path, hash);
	}
```

I think this is another opportunity to improve the documentation and examples by making it clear that collection fields of the live object have to be de-referenced from the proxy in order to have their contents persisted.

If possible, it would be good to also detect that situation at runtime and log a warning.
